actors/builtin/market/market_test.go

```diff
@@ -321,6 +321,42 @@ func TestMarketActor(t *testing.T) {
 	})
 }
 
+func TestPublishStorageDeals(t *testing.T) {
+	marketActor := tutil.NewIDAddr(t, 100)
+	owner := tutil.NewIDAddr(t, 101)
+	provider := tutil.NewIDAddr(t, 102)
+	worker := tutil.NewIDAddr(t, 103)
+	client := tutil.NewIDAddr(t, 104)
+
+	t.Run("publish a deal after activating a previous deal which has a start epoch far in the future", func(t *testing.T) {
+		startEpoch := abi.ChainEpoch(1000)
+		endEpoch := abi.ChainEpoch(2000)
+		publishEpoch := abi.ChainEpoch(1)
+
+		rt, actor := basicMarketSetup(t, marketActor, owner, provider, worker, client)
+		deal1 := generateDealProposal(client, provider, startEpoch, endEpoch)
+
+		// ensure client and provider have enough funds to lock for the deal
+		actor.addParticipantFunds(rt, client, deal1.ClientBalanceRequirement())
+		actor.addProviderFunds(rt, deal1.ProviderBalanceRequirement())
+
+		// publish the deal and activate it
+		rt.SetEpoch(publishEpoch)
+		deal1ID := actor.publishDeal(rt, deal1)
+		actor.activateDeals(rt, []abi.DealID{deal1ID}, endEpoch+1, provider)
```

Comment:
Aside: is the `+1` necessary? I think we should be allowing a sector to terminate in the same epoch as it's longest deal. Could you follow up to investigate that? Remove the +1 now if possible.

Reply:
@anorth This is done.

Reply:
I'll investigate this.